src/otx/v2/adapters/torch/mmengine/mmdet/modules/evaluation/evaluator.py

```diff
@@ -0,0 +1,469 @@
+    img_h, img_w = gt_masks.height, gt_masks.width
+    ious = bbox_overlaps(torch.from_numpy(det_bboxes), gt_bboxes.tensor, mode="iou").numpy()
+    ious[ious < iou_thr] = 0.0
+    if not ious.any():
+        return ious
+    # NOTE: further speed optimization (vectorization) could be done here
+    for coord in np.argwhere(ious):
+        m, n = coord
+        det_bbox, det_mask = sanitize_coordinates(det_bboxes[m], img_h, img_w), det_masks[m]
+        gt_bbox, gt_mask = sanitize_coordinates(gt_bboxes.numpy()[n], img_h, img_w), gt_masks[n]
+        # add padding to det_mask and gt_mask so that they have the same size
+        min_x1 = min(det_bbox[0], gt_bbox[0])
+        min_y1 = min(det_bbox[1], gt_bbox[1])
+        max_x2 = max(det_bbox[2], gt_bbox[2])
+        max_y2 = max(det_bbox[3], gt_bbox[3])
+
+        det_bbox_h, det_bbox_w = det_bbox[3] - det_bbox[1], det_bbox[2] - det_bbox[0]
+        det_mask = det_mask.resize((det_bbox_h, det_bbox_w))
+        det_mask = det_mask.expand(max_y2 - min_y1, max_x2 - min_x1, det_bbox[1] - min_y1, det_bbox[0] - min_x1)
+        gt_mask = gt_mask.crop(gt_bbox)
+        gt_mask = gt_mask.to_bitmap()
+        gt_mask = gt_mask.expand(max_y2 - min_y1, max_x2 - min_x1, gt_bbox[1] - min_y1, gt_bbox[0] - min_x1)
+        # compute iou between det_mask and gt_mask
+        det_mask = det_mask.to_ndarray()
+        gt_mask = gt_mask.to_ndarray()
+        if det_mask.shape != gt_mask.shape:
+            raise ValueDifferenceError(det_mask.shape, gt_mask.shape)
+        ious[m, n] = np.sum(det_mask & gt_mask) / np.sum(det_mask | gt_mask)
+    return ious
+
+
+def tpfpmiou_func(  # pylint: disable=too-many-locals
+    det: tuple[np.ndarray, (BitmapMasks | list)],
+    gt_masks: PolygonMasks,
+    cls_scores: np.ndarray,
+    iou_thr: float = 0.5,
+) -> tuple[np.ndarray, np.ndarray, float]:
+    """Compute tp, fp, miou for each image.
+
+    Args:
+        det (Tuple[np.ndarray, BitmapMasks]): detected bboxes and masks
+        gt_masks (PolygonMasks): ground truth polygons
+        cls_scores (np.ndarray): class scores
+        iou_thr (float): IoU threshold. Defaults to 0.5.
+
+    Returns:
+        Tuple[np.ndarray, np.ndarray, float]: tp, fp, miou for each image
+    """
+    num_dets = len(det[0])  # M
+    num_gts = len(gt_masks)  # N
+
+    tp = np.zeros(num_dets, dtype=np.float32)  # pylint: disable=invalid-name
+    fp = np.zeros(num_dets, dtype=np.float32)  # pylint: disable=invalid-name
+    gt_covered_iou = np.zeros(num_gts, dtype=np.float32)
+
+    if len(gt_masks) == 0:
+        fp[...] = 1
+        return tp, fp, 0.0
+    if num_dets == 0:
+        return tp, fp, 0.0
+
+    ious = mask_iou(det, gt_masks, iou_thr)  # (M, N)
+
+    # for each det, the max iou with all gts
+    ious_max = ious.max(axis=1)
+    # for each det, which gt overlaps most with it
+    ious_argmax = ious.argmax(axis=1)
+    # sort all dets in descending order by scores
+    sort_inds = np.argsort(-cls_scores)
```

Comment:
`sort_inds` sounds like a function name. How about `sorted_inds`?
```suggestion
    sorted_inds = np.argsort(-cls_scores)
```